Dockerfile

```diff
@@ -0,0 +1,50 @@
+FROM mcr.microsoft.com/dotnet/sdk:7.0 AS build
+WORKDIR /app
+
+COPY . .
+RUN dotnet restore
+WORKDIR ./src/Presentation
+RUN dotnet publish -c Release -o build  --no-restore --verbosity m
+
+FROM mcr.microsoft.com/dotnet/aspnet:7.0
+WORKDIR /app
+
+RUN groupadd -r -f librum-server
+RUN useradd -r -g librum-server -d /var/lib/librum-server --shell /usr/sbin/nologin librum-server
+
+
+RUN mkdir -p /var/lib/librum-server/srv
+COPY --from=build /app/src/Presentation/build/* /var/lib/librum-server/srv
+COPY --from=build /app/appsettings.json /var/lib/librum-server/srv
+RUN chmod --recursive 660 /var/lib/librum-server/
+RUN chmod 770 /var/lib/librum-server
+RUN chmod 770 /var/lib/librum-server/srv
+COPY --from=build /app/self-hosting/run.sh .
+RUN install run.sh -m770 /var/lib/librum-server/srv
+RUN rm -f /app/run.sh
+RUN chown --recursive librum-server /var/lib/librum-server/
+
+ENV JWTValidIssuer=exampleIssuer
+ENV JWTKey=exampleOfALongSecretToken
+ENV SMTPEndpoint=smtp.example.com
+ENV SMTPUsername=mailuser123
+ENV SMTPPassword=smtpUserPassword123
+ENV SMTPMailFrom=mailuser123@example.com
+ENV AdminEmail=admin@example.com
+ENV AdminPassword=strongPassword123
+ENV DBConnectionString=Server=mariadb;port=3306;Database=librum;Uid=librum;Pwd=mariadb;
+ENV CleanUrl=http://0.0.0.0
+ENV OpenAIToken=
+ENV ASPNETCORE_ENVIRONMENT=Production
+ENV DOTNET_PRINT_TELEMETRY_MESSAGE=false
+ENV LIBRUM_SELFHOSTED=true
```

Comment:
Should we move (some) of these to a default `docker-compose.yml` instead?

Reply:
Some variables are needed or else the server won't even start. I think it's fine to leave them in the Dockerfile if they absolutely need to have a default value. `LIBRUM_SELFHOSTED` for instance should always be true, right? In what case you won't be setting this to true?

People usually write their own docker-compose files. An official one can be provided as a base or as a guide, but we can't count on people setting these variables, and I think it's better if the server works out of the box with minimal configuration.

If, on the other hand, the server doesn't need these to be defined, then yes, we can remove them.

Reply:
Perhaps I chose a wrong line to comment (simply at the bottom of the ENV block) - 100% agreed about things like `LIBRUM_SELFHOSTED`, I was talking more about the SMTP or the user credentials, I would argue that baking those defaults in can lead to unsecure deployments.

Reply:
Either way I wouldn't see this as a blocker for this MR.

Reply:
Ah yes, those should probably be left blank and let the user set them as desired. Problem is, currently LibrumServer straight up crashes if these are blank. I would think most people won't have a need for confirmation emails, so a way to disable that feature would be needed.

Reply:
Same with the JWT key. If blank, Librum should probably generate a random one.

Reply:
I would leave the ones for the default admin login tho. Most other programs I've seen have a default admin login setup automatically, so it's standard.

Reply:
Default login credentials would be OK, if we were i.e. forced to change them upon first login. Baking the credentials into the image IMHO causes an unnecessary security problem for unexperienced users.

It would be much better to move this to the compose file, as that creates a clear expectation for users to change the default credentials to something else (at least until we have the CLI tool to manage users).

Reply:
Agreed. I'll change it.